src/Namicorn.ts

```diff
@@ -98,48 +99,88 @@ class Namicorn {
   /**
    * Resolves the given domain
    * @async
-   * @param domain - domain name to be resolved
-   * @returns - Returns a promise that resolves in an object
+   * @param {string} domain - domain name to be resolved
+   * @returns {Promise<NamicornResolution>} - Returns a promise that resolves in an object
    */
   async resolve(domain: string): Promise<NamicornResolution> {
     if (this.blockchain) {
       return await this.resolveUsingBlockchain(domain);
     } else {
-      const response = await myfetch(`${this.api}/${domain}`, {
-        method: 'GET',
-        headers: headers,
-      });
+      const response = await myFetch(`${this.api}/${domain}`)
       return response.json();
     }
   }
 
   /**
    * Resolves give domain name to a specific currency address if exists
    * @async
-   * @param domain - domain name to be resolved
-   * @param currencyTicker - currency ticker like BTC, ETH, ZIL
-   * @returns - A promise that resolves in an address or null
+   * @param {string} domain - domain name to be resolved
+   * @param {string} currencyTicker - currency ticker like BTC, ETH, ZIL
+   * @returns {Promise<string | null>} - A promise that resolves in an address or null
+   */
+  async address(
+    domain: string,
+    currencyTicker: string,
+  ): Promise<string | null> {
+    try {
+      return await this.addressOrThrow(domain, currencyTicker);
+    } catch (error) {
+      if (error instanceof ResolutionError) {
+        return null;
+      } else {
+        throw error;
+      }
+    }
+  }
+
+  /**
+   * Resolves given domain to a specific currency address or throws
+   * @param domain - domain name
+   * @param currencyTicker - currency ticker such as
+   *  - ZIL
+   *  - BTC
+   *  - ETH
+   * @throws UNREGISTERED_DOMAIN
+   * @throws NOT_REGISTERED_CURRENCY
```

Comment:
I don't think this is going to be a readable documentation. If you work with namicorn for the first time, you would not realize that these are codes inside `ResolutionError`. It is better to use `@throws ResolutionError` and it should throw all possible codes.